### ISSUE TITLE
Paging: incorrect behavior

### ISSUE THREAD

**Issue by NicolasSiver** [NodeBB/NodeBB#3888]
How to fix this?

By visiting latest pages, there is incorrect calculation: no posts, incorrect position, etc.

<img width="1197" alt="screen shot 2015-11-20 at 7 23 55 pm" src="https://cloud.githubusercontent.com/assets/135263/11315209/46ba461c-8fbc-11e5-8468-8ee2cb2a4b7c.png">

Core:  `NodeBB v0.8.2`
Persona: `3.0.13`

**Comment by barisusakli** [comment]
What is the topic id and database(redis/mongo) ?

**Comment by NicolasSiver** [comment]
Database: mongo
Topic Id: 1243 (why are you asking tid? )

**Comment by barisusakli** [comment]
So we can calculate the post count.

Run this in mongo

```
db.objects.count({_key:"tid:1243:posts"});
```

If the count is different from 309 run below query to update it.

```
db.objects.update({_key: "topic:1234"}, {$set: {"postcount":<value_from_count_query>}}, {upsert:true});
```

**Comment by NicolasSiver** [comment]
<img width="347" alt="screen shot 2015-11-21 at 6 32 00 pm" src="https://cloud.githubusercontent.com/assets/135263/11321040/31d44e88-907e-11e5-874d-54005428c0f9.png">

As I understand, `308` is a correct value, because one post is a part of the topic (first post)?

**Comment by barisusakli** [comment]
Hmm actually if its 308, the the postcount in topic data seems to be correct ie it should be 309 with the main post.

Is this happening on all topics or just this one?

**Comment by NicolasSiver** [comment]
@barisusakli yes.

Here is demo video: https://www.dropbox.com/s/u9po3dqcu7r1utf/nodebb-paging-issue.mov?dl=0 (download for better quality)

As you can see, I can even trigger another paging issue by going to last post

**Comment by barisusakli** [comment]
Are you filtering deleted posts server side? That might explain why there are empty pages like that. This was the reason why we don't filter them server side and just replace their content with `this post is deleted`

**Comment by NicolasSiver** [comment]
No, I don't filter deleted posts.
I don't have such plugins.

**Comment by barisusakli** [comment]
Can you put `console.log(data.posts.length)` before and after [this](https://github.com/NodeBB/NodeBB/blob/v0.8.x/src/posts.js#L70)

Then go into a topic that has the problem and let me know the output. Also is this an imported forum or a brand new one?

**Comment by NicolasSiver** [comment]
When you enter topic: 

```
21
21
```

When you go to last page:

```
1
1
```

When you go to last post (via navigation at top):

```
10
10
```

P.S.
It's imported forum.

**Comment by NicolasSiver** [comment]
Any thoughts? ;)
I have checked paging on NodeBB Community, everything works fine, seems some 0.8 related stuff that was fixed, point me in right direction (maybe you remember something that is related to paging)

For now, I would like to have solid state in `0.8.x`, and only after migrate to `0.9.x`

**Comment by barisusakli** [comment]
Does it happen if you start a new topic on your forum? I would make a copy of your database and see if it resolved in 0.9.x latest. So we know if its really a 08.x problem. There were quite a bit of changes to pagination and infinite scroll between 0.8.x and 0.9.0

**Comment by NicolasSiver** [comment]
Some sort of chain reaction...

<img width="1313" alt="screen shot 2015-11-23 at 7 07 55 pm" src="https://cloud.githubusercontent.com/assets/135263/11353929/9b16cd74-9215-11e5-9784-a90f6618bab1.png">

Trying to use my database from `0.8.x` with `0.9.x`

**Comment by NicolasSiver** [comment]
Second try to the same command `./nodebb upgrade` , and everything updated well.

<img width="894" alt="screen shot 2015-11-23 at 7 11 00 pm" src="https://cloud.githubusercontent.com/assets/135263/11353963/f8467f8a-9215-11e5-9d46-f240ff7155d3.png">

**Comment by barisusakli** [comment]
Yeah that error happens because of the number of upgrades we are trying to run in the async.series() call. I think we are hitting some sort of limit. The second time you run it it skips them all so its fine.

**Comment by NicolasSiver** [comment]
@barisusakli Paging works _well_ with `0.9.x` (with my current database), no problems with it at all.
Go to last, just page navigation, etc. 

So problem with `0.8.x`

**Comment by NicolasSiver** [comment]
Migrated to 0.9, not sure if it will be addressed for 0.8
